src/audio_sdl.cpp

```diff
@@ -125,4 +175,16 @@ void SdlAudio::UnlockMutex() const {
 #endif
 }
 
+void SdlAudio::LockMidiOutMutex() const {
+	if (SDL_LockMutex(midiout_mutex) != 0) {
+		Output::Debug("SDL_LockMutex failure: {}", SDL_GetError());
+	}
+}
+
+void SdlAudio::UnlockMidiOutMutex() const {
+	if (SDL_UnlockMutex(midiout_mutex) != 0) {
+		Output::Debug("SDL_UnlockMutex failure: {}", SDL_GetError());
+	}
+}
```

Comment:
I cannot think of any platform with MIDI support that has no working std::thread or std::mutex.
Therefore will simplify most of the logic and move it directly into AudioGeneric (or some new class like AudioGenericMidiThread).